File: LeetCode_3487/Main.java
import java.util.Arrays;

class Solution {
    public int maxSum(int[] nums) {
        Arrays.sort(nums);
        int sum = nums[nums.length - 1];
        int mx = sum;

        for (int i = nums.length - 2; i >= 0; i--) {
            if (nums[i] != nums[i + 1]) {
                sum += nums[i];
                mx = Math.max(mx, sum);
            }
        }
        return mx;
    }
}

public class Main {
    public static void main(String[] args) {
        
    }
}

File: LeetCode_3487/Main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSum = function(nums) {
    nums.sort((a, b) => a - b);

    let sum = nums[nums.length - 1];
    let mx = sum;
    for (let i = nums.length - 2; i >= 0; i--) {
        if (nums[i] != nums[i + 1]) {
            sum += nums[i];
            mx = Math.max(mx, sum);
        }
    }
    return mx;
};